lib/Sema/TypeCheckAttr.cpp

```diff
@@ -6910,6 +6910,26 @@ void AttributeChecker::visitNonisolatedAttr(NonisolatedAttr *attr) {
           return;
         }
 
+        if (attr->isUnsafe() && type->isSendableType() && var->isStatic() &&
```

Comment:
It's not necessary to limit this check to static variables. You can write `nonisolated(unsafe)` on any kind of storage, including stored properties and local variables, and it'll always be unnecessary if the variable is immutable and `Sendable`